ctapipe/reco/tests/test_HillasReconstructor.py

```diff
@@ -105,54 +117,190 @@ def test_reconstruction():
 
     reconstructed_events = 0
 
+    # ==========================================================================
+
     for event in source:
         calib(event)
 
         mc = event.simulation.shower
         array_pointing = SkyCoord(az=mc.az, alt=mc.alt, frame=horizon_frame)
-        hillas_dict = {}
+
+        hillas_dict_CameraFrame = {}
+        hillas_dict_TelescopeFrame = {}
         telescope_pointings = {}
-        for tel_id, dl1 in event.dl1.tel.items():
 
-            geom = source.subarray.tel[tel_id].camera.geometry
+        for tel_id, dl1 in event.dl1.tel.items():
 
             telescope_pointings[tel_id] = SkyCoord(
                 alt=event.pointing.tel[tel_id].altitude,
                 az=event.pointing.tel[tel_id].azimuth,
                 frame=horizon_frame,
             )
+
+            geom_CameraFrame = source.subarray.tel[tel_id].camera.geometry
+
+            # this could be done also out of this loop,
+            # but in case of real data each telescope would have a
+            # different telescope_pointing
+            geom_TelescopeFrame = geom_CameraFrame.transform_to(
+                TelescopeFrame(telescope_pointing=telescope_pointings[tel_id])
+            )
+
             mask = tailcuts_clean(
-                geom, dl1.image, picture_thresh=10.0, boundary_thresh=5.0
+                geom_TelescopeFrame,
+                dl1.image,
+                picture_thresh=10.0,
+                boundary_thresh=5.0,
```

Comment:
All of the beginning of this test can be replaced with an ImageProcessor (assuming it also is modified to use the TelescopeFrame geometries).  The full test should be:

```py3
for event in source:
    calib(event)
    process_images(event)
    reconstruct(event)
    check_reconstruction(event)
```

That will clean things up a lot.